### DISCUSSION TITLE
[Blog Plugin] Way to fill sub-pages with posts based on categories, tags, etc?

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#5821] [Ideas]
This may be a bit much of an idea for the Blog plugin, but how about a system that would allow you to list posts based on conditions within a sub-page?

Let me try to explain:

Imagine having the following folder structure:
```
.
├── docs/
│   ├── posts/
│   │   ├── 2022/
│   │   │   └── 01-01-some-post.md
│   │   └── 2023/
│   │       └── 06-10-another-post.md
│   └── index.md
└── mkdocs.yml
```
This would, when generated, result in paths such as `/2022/01-01-some-post/` or `/2023/06-10-another-post/`
The main issue with this is, that a path such as `/2022/` results in a 404 since there is no index page to get.

So my proposal is to allow setting index pages that display the posts belonging into this sub-page (or maybe based on criterias).

I could think of two options:

- Config option to auto-generate indexes where pages are missing within the posts directory.
- Implement the same behaviour as with the main index page, meaning injecting the posts into the index page when present. And if none is present do nothing.

I hope it's understandable what my goal is?
I effectively ask for a way within the built-in blog plugin to turn this final structure
```
blog.example.com/
├── 2022/
│   └── 01-01-some-post/
│       └── index.html
├── 2023/
│   └── 06-10-another-post/
│       └── index.html
└── index.html
```
into this structure:
```
blog.example.com/
├── 2022/
│   ├── index.html
│   └── 01-01-some-post/
│       └── index.html
├── 2023/
│   ├── index.html
│   └── 06-10-another-post/
│       └── index.html
└── index.html
```
Adding index pages for the sub-folders without any content that would display posts within sub-folders there...

**Comment by squidfunk** [comment]
Already requested in #5074 and on the roadmap for Insiders.
